### PR TITLE
Collections girder ansible module

### PR DESCRIPTION
**Pull request by kotfic**
Adds support for idempotent creation of ```collection```, ```folder```, ```item```, ```file``` resources as well as basic ```upload``` functionality to the girder ansible module.

```yaml
    - name: Create collection
      girder:
        port: 8080
        username: "admin"
        password: "letmein"
        collection:
          name: "Test Collection"
          description: "A test collection"
      register: test_collection

    - name: Create folder
      girder:
        port: 8080
        username: "admin"
        password: "letmein"
        folder:
          parentType: "collection"
          parentId: "{{test_collection['gc_return']['_id'] }}"
          name: "test folder"
          description: "A test folder"
      register: test_folder

    - name: Create an item
      girder:
        port: 8080
        username: "admin"
        password: "letmein"
        item:
          folderId: "{{test_folder['gc_return']['_id'] }}"
          name: "test item"
          description: "A test item"
      register: test_item

    - name: Upload files
      girder:
        port: 8080
        username: "admin"
        password: "letmein"
        files:
          itemId: "{{ test_item['gc_return']['_id'] }}"
          sources:
            - /tmp/data/test1.txt
            - /tmp/data/test2.txt
      register: retval
```

Uploads are guaranteed to be idempotent based on file name (basename) and size. In all cases you may add ```state: absent``` to ensure the resource (e.g.,  collection, folder, item, file) is not present. 

Still left:
- [x] Add access control
  - [x] Create groups
  - [x] Set access on resource for users, groups
- [X] Provide hierarchical processing so ```collection``` can include ```folders``` section, folders can include items, items,  files etc. 
- [ ] Testing (i want to include automated testing through Travis)
- [x] Documentation

### CODE REVIEW

**Comment by zachmullen** [comment]
@kotfic it seems something related to the ansible module is breaking our RTD build. The command that causes the failure is

    python setup.py install --force

and the error is:

```
linking /home/docs/checkouts/readthedocs.org/user_builds/girder/envs/latest/lib/python2.7/site-packages/girder/clients/ansible/test/library/girder.py -> ../../girder.py
error: [Errno 17] File exists
```

I am able to recreate this locally if I run that command twice in a row. Any idea how to fix that?

**Comment by zachmullen** [comment]
Not sure the underlying reason for the failure, but #1196 fixes that breakage by simply not packaging the test file that's causing the break.

**Comment by zachmullen** [comment]
@kotfic do you think we could get this merged in the near future? It's got some great features in here, I think if we could get them some basic documentation that would be sufficient to merge this branch.

**Comment by kotfic** [comment]
@zachmullen yes,  sorry -  this has been on the backburner for some time.  I was hoping to get CI testing,  but I can put together some documentation either this weekend,  or next week.

**Comment by zachmullen** [comment]
Awesome :smile:

**Comment by kotfic** [comment]
@zachmullen Here is that documentation -  I am seeing a bug with my test_resources.yml  playbook.  I want to track that down before you review.

**Comment by zachmullen** [comment]
Documentation looks good, once you figure out that bug and merge master this should be ready.

File: clients/ansible/girder.py [hunk 1]
```diff
@@ -310,6 +460,86 @@
...
+    password: "letmein"
+    user:
+      firstName: "Alice"
+      lastName: "Test"
+      login: "alice"
+      password: "letmein"
+      email: "alice.test@kitware.com"
+    state: present
+
+# Create an 'bill' user
```
Comment (zachmullen) [inline clients/ansible/girder.py]: a*

File: clients/ansible/girder.py [hunk 2]
```diff
@@ -310,6 +460,86 @@
...
+    password: "letmein"
+    user:
+      firstName: "Bill"
+      lastName: "Test"
+      login: "bill"
+      password: "letmein"
+      email: "bill.test@kitware.com"
+    state: present
+
+# Create an 'chris' user
```
Comment (zachmullen) [inline clients/ansible/girder.py]: a*

File: clients/ansible/girder.py [hunk 3]
```diff
@@ -372,6 +602,154 @@
...
+    item:
+      folderId: "{{test_folder['gc_return']['_id'] }}"
+      name: "test item"
+      description: "A test item"
+  register: test_item
+
+# Upload files on the localhost at /tmp/data/test1.txt and
+# /tmp/data/test2.txt to the girder instance under the item
+# "test item"
+# Note:  the list is idempotent and will remove file that are
```
Comment (zachmullen) [inline clients/ansible/girder.py]: files*

**Comment by zachmullen** [comment]
LGTM

**Pull request merged**